client/components/test-mode-notice/index.tsx

```diff
@@ -0,0 +1,180 @@
+								'woocommerce-payments'
+							),
+							'WooPayments'
+						),
+						components: {
+							strong: <strong />,
+						},
+					} ) }
+				</>
+			) : (
+				<>
+					{ interpolateComponents( {
+						mixedString: sprintf(
+							/* translators: %1$s: WooPayments */
+							__(
+								'{{strong}}%1$s is in test mode.{{/strong}} All transactions will be simulated. {{learnMoreLink}}Learn more{{/learnMoreLink}}',
+								'woocommerce-payments'
+							),
+							'WooPayments'
+						),
+						components: {
+							strong: <strong />,
+							learnMoreLink: (
+								// Link content is in the format string above. Consider disabling jsx-a11y/anchor-has-content.
+								// eslint-disable-next-line jsx-a11y/anchor-has-content
+								<Link
+									href={
+										'https://woo.com/document/woopayments/testing-and-troubleshooting/dev-mode/'
+									}
+									target="_blank"
+									rel="noreferrer"
+								/>
+							),
+						},
+					} ) }
+				</>
+			);
+		case 'documents':
+		case 'deposits':
+		case 'disputes':
+		case 'payments':
+		case 'loans':
+		case 'transactions':
+			return isDetailsView ? (
+				<>
+					{ interpolateComponents( {
+						mixedString: sprintf(
+							/* translators: %1$s: WooPayments */
+							_n(
+								'%1$s was in test mode when this %2$s was %3$s. To view live %2$ss, disable test mode in {{settingsLink}}%1$s settings{{/settingsLink}}.',
+								'%1$s was in test mode when these %2$ss were %3$s. To view live %2$ss, disable test mode in {{settingsLink}}%1$s settings{{/settingsLink}}.',
+								'deposits' === currentPage ? 2 : 1,
+								'woocommerce-payments'
+							),
+							'WooPayments',
+							nounToUse[ currentPage ],
+							verbToUse[ currentPage ]
+						),
+						components: {
+							settingsLink: (
+								// Link content is in the format string above. Consider disabling jsx-a11y/anchor-has-content.
+								// eslint-disable-next-line jsx-a11y/anchor-has-content
+								<a href={ getPaymentSettingsUrl() } />
+							),
+						},
+					} ) }
+				</>
+			) : (
+				<>
+					{ interpolateComponents( {
+						mixedString: sprintf(
+							/* translators: %1$s: WooPayments */
+							__(
+								'Viewing test %1$s. To view live %1s, disable test mode in {{settingsLink}}%2s settings{{/settingsLink}}.',
+								'woocommerce-payments'
+							),
+							currentPage,
+							'WooPayments'
+						),
+						components: {
+							settingsLink: (
+								// Link content is in the format string above. Consider disabling jsx-a11y/anchor-has-content.
+								// eslint-disable-next-line jsx-a11y/anchor-has-content
+								<a href={ getPaymentSettingsUrl() } />
+							),
+						},
+					} ) }
+				</>
+			);
+	}
+};
+
+export const TestModeNotice: React.FC< Props > = ( {
+	currentPage,
+	actions,
+	isDetailsView = false,
+	isDevMode = false,
+} ) => {
+	if ( ! isInTestMode() ) return null;
+
+	return (
+		<BannerNotice
+			status="warning"
+			icon={ false }
```

Comment:
There is no need to set the icon to false, it behaves like that by default 🙂

Reply:
Good to know! Updated in 890dd09f5.